server/src/test/java/org/elasticsearch/repositories/blobstore/ShardSnapshotTaskRunnerTests.java

```diff
@@ -124,7 +124,15 @@ public static SnapshotShardContext dummyContext(final SnapshotId snapshotId, fin
             IndexMetadata.builder(indexId.getName()).settings(indexSettings(IndexVersion.current(), 1, 0)).build(),
             Settings.EMPTY
         );
-        final var dummyStore = new Store(shardId, indexSettings, new ByteBuffersDirectory(), new DummyShardLock(shardId));
+        final var dummyStore = new Store(
+            shardId,
+            indexSettings,
+            new ByteBuffersDirectory(),
+            new DummyShardLock(shardId),
+            Store.OnClose.EMPTY,
+            false,
+            false
+        );
```

Comment:
This is one such case, alternatively we could add it to a collection to be closed upon `@After`.